addons/hr_expense/tests/test_expenses.py

```diff
@@ -1753,3 +1753,28 @@ def test_expense_product_update(self):
             {'name': 'test sheet no update', 'unit_amount': 100.0, 'quantity': 1, 'total_amount': 100.0},
             {'name':    'test sheet update', 'unit_amount': 250.0, 'quantity': 1, 'total_amount': 250.0},  # no update
         ])
+
+    def test_foreign_currencies_total(self):
+        foreign_currency = self.env['res.currency'].create({
+            'name': 'DBL',
+            'symbol': 'DD',
+            'rounding': 0.01,
+            'currency_unit_label': 'Doubloon'
+        })
+        Expense = self.env['hr.expense'].with_user(self.expense_user_employee)
+        Expense.create({
+            'name': 'Company expense',
+            'payment_mode': 'company_account',
+            'total_amount': 1000.00,
+            'employee_id': self.expense_employee.id,
+        })
+        Expense.create({
+            'name': 'Company expense 2',
+            'payment_mode': 'company_account',
+            'currency_id': foreign_currency.id,
+            'total_amount': 1000.00,
+            'total_amount_company': 2000.00,
+            'employee_id': self.expense_employee.id,
+        })
```

Comment:
Always better to batch when possible
```suggestion
        Expense.create([{
            'name': 'Company expense',
            'payment_mode': 'company_account',
            'total_amount': 1000.00,
            'employee_id': self.expense_employee.id,
        }, {
            'name': 'Company expense 2',
            'payment_mode': 'company_account',
            'currency_id': foreign_currency.id,
            'total_amount': 1000.00,
            'total_amount_company': 2000.00,
            'employee_id': self.expense_employee.id,
        }])
```